features/steps/thens/alignment.py

```diff
@@ -61,15 +61,20 @@ def count_segments(logic, representation):
     and geometry representation.
     """
     try:
-        logic_count = len(logic.segments)
+        expected_count = 0
+        for seg in logic.segments:
+            if seg.PredefinedType == "HELMERTCURVE":
+                expected_count += 2
+            else:
+                expected_count += 1
     except AttributeError:
```

Comment:
Later we do `(logic_only or rep_only)`. So this seems to imply that the try-except block is to catch `logic.segments`. But the block also catching `seg.PredefinedType` so if there is an unexpected instance there without a predefined type attribute, the loop is terminated and the segments that were presented are not considered. This is not really a concern (because we have schema validation in place anyway), it's just that I find the intention behind try catch a bit unclear.

I would rather see `if logic.has_segments:` which makes it much more explicit which cases we're expecting and handling. But for me it's not necessary to address this now.

Reply:
I went ahead and revised this while still fresh in my mind.